docs/hooks/hook_civicrm_alterUFFields.md

```diff
@@ -0,0 +1,37 @@
+# hook_civicrm_alterUFFIelds
+
+## Summary
+
+This hook allows for the modification of the avaliable fields that are permissiable for use within a profile.
+
+## Definition
+
+    hook_civicrm_alterUFFIelds(&$fields) {
+
+## Parameters
+
+-   $fields - a multidimential array of Entities and their fields.
+      For each field contained within the array there are at the following properties
```

Comment:
@seamuslee001 which, if any, of these properties are used?

Reply:
As far as I know all are I believe

Reply:
I know for certain that the name is and i think others are used but i think the general gist is that your generating the same output as from a `entity.getfields` api call